types/office-js/index.d.ts

```diff
@@ -163,78 +289,101 @@ declare namespace Office {
-        * @param startAddress Accepts the initial HTTPS Url that opens in the dialog.
-        */
-        displayDialogAsync(startAddress: string): void;
-        /**
-        * Displays a dialog to show or collect information from the user or to facilitate Web navigation.
-        * @param startAddress Accepts the initial HTTPS Url that opens in the dialog.
-        * @param options Optional. Accepts a DialogOptions object to define dialog behaviors.
-        */
-        displayDialogAsync(startAddress: string, options: DialogOptions): void;
-        /**
-        * Displays a dialog to show or collect information from the user or to facilitate Web navigation.
-        * @param startAddress Accepts the initial HTTPS Url that opens in the dialog.
-        * @param callback Optional. Accepts a callback method to handle the dialog creation attempt.
-        */
-        displayDialogAsync(startAddress: string, callback: (result: AsyncResult) => void): void;
-        /**
-        * Displays a dialog to show or collect information from the user or to facilitate Web navigation.
-        * @param startAddress Accepts the initial HTTPS Url that opens in the dialog.
-        * @param options Optional. Accepts a DialogOptions object to define dialog behaviors.
-        * @param callback Optional. Accepts a callback method to handle the dialog creation attempt.
+        * Displays a dialog to show or collect information from the user or to facilitate Web navigation. 
+		*         
+        * @remarks
+        * Hosts: Word, Excel, Outlook, PowerPoint
+        * 
+        * Requirement sets: DialogApi, Mailbox 1.4
+        * 
+        * Last changed in: DialogApi 1.1, Mailbox 1.4
+        * 
+        * The initial page must be on the same domain as the parent page (the startAddress parameter). After the initial page loads, you can go to other domains.
+        * 
+        * Any page calling office.context.ui.messageParent must also be on the same domain as the parent page.
+        * 
+        * The following design considerations apply to dialog boxes:
+        * 
+        * - An Office Add-in can have only one dialog box open at any time.
+        * 
+        * - Every dialog box can be moved and resized by the user.
+        * 
+        * - Every dialog box is centered on the screen when opened.
+        * 
+        * - Dialog boxes appear on top of the host application and in the order in which they were created.
+        * 
+        * Use a dialog box to:
+        * 
+        * - Display authentication pages to collect user credentials.
+        * 
+        * - Display an error/progress/input screen from a ShowTaspane or ExecuteAction command.
+        * 
+        * - Temporarily increase the surface area that a user has available to complete a task.
+        * 
+        * Do not use a dialog box to interact with a document. Use a task pane instead.
```

Comment:
Importantly, they actually won't let you!  You can't call document-manipulating APIs from within dialogs.